certificates.go

```diff
@@ -101,6 +103,43 @@ func (c *CertificatesServiceOp) List(ctx context.Context, opt *ListOptions) ([]C
 	return root.Certificates, resp, nil
 }
 
+func (c *CertificatesServiceOp) ListByName(ctx context.Context, name string, opt *ListOptions) (*Certificate, *Response, error) {
+
+	if len(name) < 1 {
+		return nil, nil, NewArgError("name", "cannot be an empty string")
+	}
+
+	path := fmt.Sprintf("%s?name=%s", certificatesBasePath, name)
+	urlStr, err := addOptions(path, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := c.client.NewRequest(ctx, http.MethodGet, urlStr, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(certificatesRoot)
+	resp, err := c.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	if l := root.Links; l != nil {
+		resp.Links = l
+	}
+	if m := root.Meta; m != nil {
+		resp.Meta = m
+	}
+
+	if len(root.Certificates) == 0 {
+		return nil, resp, err
+	}
+
+	return &root.Certificates[0], resp, err
```

Comment:
Minor nitpick for consistency. Our list methods generally return an empty array rather than `nil` when there are no results. We're also returning a single cert rather than an array. Maybe it would be rename this to `GetByName` and not take a `ListOptions` as an argument if we're not actually going to return a list here?

Reply:
I think lets keep the listoption if in future we want to expand the endpoints and its support. I will modify the changes with the return slice & if nothing is found then empty slice

Reply:
done. please check